src/jquery/core.js

```diff
@@ -11,68 +11,87 @@ var findProp,
 	class2type = {},
 	oldInit = jQuery.fn.init,
 	oldFind = jQuery.find,
-
 	rattrHashTest = /\[(\s*[-\w]+\s*)([~|^$*]?=)\s*([-\w#]*?#[-\w#]*)\s*\]/,
 	rattrHashGlob = /\[(\s*[-\w]+\s*)([~|^$*]?=)\s*([-\w#]*?#[-\w#]*)\s*\]/g,
 
 	// Require that the "whitespace run" starts from a non-whitespace
 	// to avoid O(N^2) behavior when the engine would try matching "\s+$" at each space position.
 	rtrim = /^[\s\uFEFF\xA0]+|([^\s\uFEFF\xA0])[\s\uFEFF\xA0]+$/g;
 
-migratePatchFunc( jQuery.fn, "init", function( arg1 ) {
-	var args = Array.prototype.slice.call( arguments );
-
-	if ( jQuery.migrateIsPatchEnabled( "selector-empty-id" ) &&
-		typeof arg1 === "string" && arg1 === "#" ) {
-
-		// JQuery( "#" ) is a bogus ID selector, but it returned an empty set
-		// before jQuery 3.0
-		migrateWarn( "selector-empty-id", "jQuery( '#' ) is not a valid selector" );
-		args[ 0 ] = [];
-	}
+migratePatchFunc(
+	jQuery.fn,
+	"init",
+	function( arg1 ) {
```

Comment:
I'd like to avoid all this reformatting in this file. We don't use Prettier and any time we do major code style changes like that - especially ones modifying line numbers - it gets harder to look at history.

If something is a result of stricter ESLint then fine, but this looks excessive to me.

Reply:
I don't really know of a better way to format this. I understand the concern about diffing, but I think the way Prettier formatted this is the way it should have been formatted from the beginning. And, doing it now makes it less likely we'll have to make more formatting changes in the future. It was ESLint that prompted this, but I prefer to let tools do the formatting as much as possible so that maintainers like us don't waste any time doing delicate dances to get the code to pass lint.

Reply:
Well, there is also the current style. Right now, this PR splits the style as other files use what was here before; you'd have to reformat everything to make it consistent again.

What exactly in ESLint triggered these changes?

> I prefer to let tools do the formatting as much as possible so that maintainers like us don't waste any time doing delicate dances to get the code to pass lint

I have a more nuanced view on this. I do use formatters a lot - and usually it's Prettier as well - as it helps with avoiding some discussions, especially in large teams. But integrating such a tool doesn't mean you stop having formatting changes, unfortunately. Prettier's GitHub has many discussions about how to best format various code constructs and they change things from time to time. In my experience, having Prettier in a project means that you periodically need to update it & then reformat a huge percentage of files, creating huge diffs and obstructing the Git history. In larger teams & projects I believe it's still worth it but I have doubts about jQuery with just a few regular committers and quite rare source changes. In some files, we'd end up with the only modifications being occasional Prettier runs.

One option is to employ usage of https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view, although I expect it's not perfect and won't erase all the noise.

As for the current Migrate style as it's on `main`, I actually like it as the following:
```js
migratePatchFunc( jQuery.fn, "init", function( arg1 ) {
	/* code */
}, "selector-empty-id" );
```
is quite similar visually to:
```js
jQuery.fn.init = function( arg1 ) {
	/* code */
};
```
which is mostly what happens, modulo the `migratePatchFunc` wrapper that is needed for being able to selectively disable patches.

Reply:
The first is what was failing. FWIW, I disagree on which is better. In the first example, it's easy to overlook that last argument after the function, especially when the function has enough lines. Putting each argument on its own line is the clearest way to prevent that. And, there are several cases in this file where there are 2 string arguments after the function, exacerbating the problem.

But, if you're adamant, we can avoid changing the whole file. This all started because you preferred to not change the indentation in the one of the functions that was inconsistent with the others, but had a ternary return that made it look awkward to be consistent. A possible solution for that is to do what was done elsewhere and start the function declaration on the next line. See the latest.